### ISSUE TITLE
Could not find image named undefined.png

### ISSUE THREAD

**Issue by kirkstrobeck** [wcandillon/react-native-img-cache#17]
Getting many of these warnings. Images still show though. Don’t get this with `Image`.

```bash
Could not find image named undefined.png
```

**Comment by wcandillon** [comment]
@kirkstrobeck could you send me a snippet that would allow me to reproduce this bug?

**Comment by kirkstrobeck** [comment]
I’m not sure that I can without hours of work. Feel free to close as it only produces a warning. Perhaps there can be a check that somewhere says *thing* must not be `undefined`. There is a guard around the image iterator that ensures there is a proper url on my end.

**Comment by wcandillon** [comment]
@kirkstrobeck Any change you could send me the props to `CachedImage`? What is undefined in the props you send to the component?

**Comment by kirkstrobeck** [comment]
Here is an example. Unfortunately, nothing is undefined. `hasImage1` ensures `photos[1]` is available.

```js
{
  hasImage1
  && <CachedImage
    blurRadius={30}
    source={{
      uri: photos[1],
    }}
    style={{
      bottom: 0,
      left: 0,
      position: 'absolute',
      right: 0,
      top: 0,
      zIndex: 9,
    }}
  />
}
```

**Comment by wcandillon** [comment]
@kirkstrobeck and `photos[1]` is `undefined`?

**Comment by kirkstrobeck** [comment]
No, `*.png` isn’t specified in my code. The ajax data gets `.jpg` files. I have a couple local `.png` files that I was running thru `CachedImage`, but it wasn’t working so now I run them thru `Image`. Perhaps uninstalling the app would blow away the cache? but that image isn’t referenced any longer so not sure whats causing this? Looks like the pattern is that whenever I use `CachedImage` it warns about an error case from the past and doesn’t forget.

**Comment by wcandillon** [comment]
@kirkstrobeck I ran a couple of tests on IOS and couldn't reproduce the issue. Is it possible that you experienced the issue on android? Please feel free to re-open if this is the case of if you have more information on how to reproduce the issue.

**Comment by kirkstrobeck** [comment]
My hunch was correct. Removing the app and reinstalling resolved the error as it wiped the cache.

**Comment by dillonykyang** [comment]
@wcandillon I am also seeing this issue on iOS... just using 
<CachedImage source={{uri: item.image_url}} style={{flex:1, resizeMode: 'stretch'}} />
the image_url has a png extension and guaranteed to exist.

**Comment by dillonykyang** [comment]
@wcandillon further debugging - the warning is on only when you provide a style prop to it.

**Comment by wcandillon** [comment]
@dillonykyang @kirkstrobeck The issue should be fixed in `1.1.7`. Please keep me posted and thank you so much for your help.

**Comment by kirkstrobeck** [comment]
Upgraded, but it seems to still be present. 😭

**Comment by mstllc** [comment]
Yeah I'm seeing this issue as well and I think it's because of this line: https://github.com/wcandillon/react-native-img-cache/blob/master/src/index.tsx#L101

I actually see `render()` getting called twice before the component finally has a value at `this.state.path`, so the built in `Image` component is throwing this warning when it can't load `undefined.png`.  I have to think it's just because `RNFetchBlob.fs.exists()` is a Promise and it's delaying anything until the next loop, which is going to miss that first `render()` call?

For what it's worth, I don't think the `style` prop has anything to do with this issue as mentioned above.  All that does is allow RN to determine it doesn't even need to attempt to load the image because it has no dimensions and won't be displayed anyways, so the warning is never thrown.

I'm not exactly sure what the right solution is here, but one idea is to allow for passing a placeholder image path that could be set as the initial value of `this.state.path` so that something is displayed while it pulls the image from cache or downloads the image.

For those who are affected by the warning, you can disable that specific yellow box warning with `console.ignoredYellowBox = ['Could not find image']` somewhere near the entry point of your app.  This will prevent the yellow box from being displayed, but you'll still see the warnings in your debugging console.

**Comment by wcandillon** [comment]
@thorsonmscott Thank you so much for providing this information, I will look into it.

**Comment by wcandillon** [comment]
@thorsonmscott @kirkstrobeck Can you check version 1.1.8?

**Comment by kirkstrobeck** [comment]
Good for a while, then many warnings like

```
Could not find image named undefined/Users/Foo/Library/Developer/CoreSimulator/Devices/AB229581-586B-40CF-864C-09A66A9A13BF/data/Containers/Data/Application/8C807149-EE98-413A-9DD3-1D5EE79FF3D5/Library/Caches/b253a77b492fc10de6c4c2075c5c17bc2ddee4cd.jpg
```
This is a regression, because when warnings show, the images do not come thru, so now it’s breaking.

Using `1.1.8`, `git clean -xfd` and a simulator reset content and settings iOS.

**Comment by wcandillon** [comment]
@kirkstrobeck should be fixed in `1.1.9`, sorry about this regression!

**Comment by kirkstrobeck** [comment]
No worries, that’s what software does. `1.1.9` appears to be fixed. 🎉
